### ISSUE TITLE
SA1100 DoNotPrefixCallsWithBaseUnlessLocalImplementationExists has incoherent behavior with Generic base class

### ISSUE THREAD

**Issue by kev-in-shu** [DotNetAnalyzers/StyleCopAnalyzers#2909]
Some invocations are marked with rule SA1100, but others are ignored:
* base.Mode
* base.TestNoError

In our CI, we have seen random behaviors with the ignored rules sometimes triggering.

Base:
    
```csharp
namespace FailingRuleSA1100
{
	public class FailSA1100 : FailSA1100Base<int, int>
	{
		private FailSA1100()
			: base("SubClassWithoutOverride")
		{
		}

		public void Init()
		{
			// There should be a Warning SA1100 here
			base.Mode = 1;

			// 2 issues with the line that invokes TestNoError,
			//   * the rule stops triggering on property Mode in the whole file
			//   * the rule does not work on this line itself
			// Comment the line to see warnings appear on the Mode property usage
			base.TestNoError(@"a");
			base.NoError(@"a");
			base.FullTest(@"a");
		}

		public void Pass()
		{
			// There should be a Warning SA1100 here
			base.Mode = 1;
		}
	}
}
```

Derived:
    
```csharp
       namespace FailingRuleSA1100
       {
           public class FailSA1100 : FailSA1100Base<int, int>
           {
               private FailSA1100()
                   : base("SubClassWithoutOverride")
               {
               }
               public void Init()
               {
                   // There should be a Warning SA1100 here
                   base.Mode = 1;

                   // 2 issues with the line that invokes TestNoError,
                   //   * the rule stops triggering on property Mode in the whole file
                   //   * the rule does not work on this line itself
                   // Comment the line to see warnings appear on the Mode property usage
                   base.TestNoError(@"a");
                   base.NoError(@"a");
                   base.FullTest(@"a");
               }
               public void Pass()
               {
                   // There should be a Warning SA1100 here
                   base.Mode = 1;
               }
           }
       }
```

Above examples are also available on https://github.com/kev-in-shu/stylecop-sa1100-debug

**Comment by bjornhellander** [comment]
I can confirm that this did not work correctly in version **1.0.2**, which the example code used.
It does however work in both 1.1.118 and the latest beta, so I suggest to close this.
